### ISSUE TITLE
Idea for a possibly improved turret system 

### ISSUE THREAD

**Issue by Arutus** [tsunamayo/Starship-EVO#2760]
Hello this is a suggestion I thought of while designing battleship turrets for a cruiser of mine. The current Addons can interfere with block turret gun elevations and generally look quite odd with them being applied directly to the weapon itself, so in an effort to remedy this id like to suggest a system more akin to the ADV cannon systems implemented in FTD in which the firing piece ( the weapon itself in SEVO) the mantlet, the Hinge block added to the front of the firing piece and addons which are adjusted to be added to the back of the firing piece as to have them hidden in the turret. 
For reference here is a link (https://steamcommunity.com/sharedfiles/filedetails/?id=54682645) to a guide behind setting up ADV cannons with the goal being understanding how the mantlet, firing piece and gauge increasers/cooling interreact and how it can be applied to the systems we have for turrets in SEVO.

**Comment by tsunamayo** [comment]
Hi, for some reason the link is not working, so it is quite hard for me to understand your suggestion having not played FTL. Cheers

**Comment by Arutus** [comment]
try copy pasting it its a link to a steam guide for FTD adv cannons

**Comment by TIKIRobo** [comment]
That would be cool, i love the turrets from FTD (from the depths)  and I think that this could really improve the way we make turrets but also make it harder to spam turrets all over your ships.

**Comment by tsunamayo** [comment]
Okay I dont really understand that guide, it seems too complex of a solution. Not sure what you dont like and what you would like to see instead.

**Comment by Arutus** [comment]
ok so what I suggested out of that whole guide where 3 systems
the firing piece as the weapon, the mantlet as a special hinge and the addon the gauge parts

**Comment by Arutus** [comment]
I felt that with this system you could make more compact turret setups than having to worry about putting addons on an extended weapon mounted directly on the hinge instead having the weapon and barrels be independant so the barrels are just that and the weapon is more of the gun breech

**Comment by Arutus** [comment]
So I suppose it would be a way to have the mantlet serve as a connection between the barrels the weapon (firing piece in the guide) and still have them elevate independent from the weapon with the only concern after just fitting the addons in the turret itself instead of the barrel

**Comment by tsunamayo** [comment]
I am sorry I still have no idea what you are suggesting ;) I guess you use too much FTD vocabulary, mantle, gauge. Even the hinge, are you referring to starship EVO hinge or something else?

**Comment by Arutus** [comment]
no I mean the blocks in the guide
im comparing them to the equivalent blocks in SEVO

**Comment by Arutus** [comment]
in the guide he labels each part I just compared them to the ones in SEVO with the mantlet being the SEVO hinge the Firing Piece The weapon block/brick and the barrel being just that

**Comment by Arutus** [comment]
just focus on those 3 parts and look at the turret set ups at the bottom

**Comment by Arutus** [comment]
it should explain how the system would work if you look for only those parts in each set up

**Comment by TIKIRobo** [comment]
I dont exactly know if this is what Artutus means but hopefully my bad drawing will hep
1. this is the gun itself so its the beam weapon or laser weapon 
2 this is a hinge 
3 this would be a new block that you could attach something like the charge addon onto instead of directly putting it on the beam/laser weapon block but it would still make the weapon have the effects of the charge addon then  you could put rotator under this new block and then build the rest of the turret (ie the good looking bits) around the new block and the gun 
hope this helps!
![BAD drawing](https://user-images.githubusercontent.com/56371294/89699628-0e72d480-d8dd-11ea-9aee-907d8f99424e.png)

**Comment by Arutus** [comment]
yes that's exactly it so you could hide and protect the addons while designing the turret without restrictions

**Comment by Arutus** [comment]
Oh seems the thread has lost traction
